kore/src/Kore/ASTVerifier/PatternVerifier.hs

```diff
@@ -358,15 +365,18 @@ verifyPatternHead (_ :< patternF) =
         Syntax.TopF top ->
             transCofreeF Internal.TopF <$> verifyTop top
         Syntax.VariableF var ->
-            transCofreeF (Internal.VariableF . getConst)
-                <$> verifyVariable var
+            transCofreeF (wrapVariable . getConst)
+                <$> verifyVariable (RegVar var)
         Syntax.InhabitantF _ ->
             koreFail "Unexpected pattern."
         Syntax.SetVariableF (SetVariable var) ->
-            transCofreeF (Internal.SetVariableF . SetVariable . getConst)
-                <$> verifyVariable var
+            transCofreeF (wrapVariable . getConst)
```

Comment:
Could you add a test similar to `Object pattern - different variable sort`, but for set variables?